### PR TITLE
ensure nothing is staged as part of undoing the first commit

### LINKED ISSUE #2656: Ignoring files on an unborn branch
When a user uses `right-click` to ignore (one or more) files on an unborn branch, the files do not "disappear" from the Changes tab as they are supposed (under a Publish repo). Only upon commit of the `.gitignore` file, do the ignored files "disappear".

Note, this applies to both `Ignore` and `Ignore all x Files` options upon right-clicking. Not 100% of what the correct outcome should be.

Reproduce:
- Create new repo
- undo first commit
- try to ignore a file other than `gitignore`
- if you commit only `gitignore`, the other files are removed on Changes tab

screen shot 2017-09-08 at 1 09 04 pm
screen shot 2017-09-08 at 1 12 53 pm

tested in 0.8.1-beta4. mac osx.

### PR DESCRIPTION
**Pull request by shiftkey**
Fixes #2656

 - [x] documentation
 - [x] can we do this more elegantly?

### CODE REVIEW

File: app/src/lib/git/update-index.ts [hunk 1]
```diff
@@ -157,3 +157,21 @@ export async function stageFiles(
...
+  files: ReadonlyArray<WorkingDirectoryFileChange>
+): Promise<void> {
+  for (const file of files) {
+    const paths = []
+    paths.push(file.path)
+    if (file.oldPath) {
+      paths.push(file.oldPath)
+    }
+
+    await updateIndex(repository, paths, { forceRemove: true })
```
Comment (shiftkey) [inline app/src/lib/git/update-index.ts]: I like using `update-index` here as we're already using this elsewhere, but it requires multiple invocations (no way to recurse or pass in a wildcard expression). `git rm --cached -r .` is a neat shortcut and handles recursion, but for most people the initial commit will be just a few files, so it's not a huge win.  @desktop/core any strong feels on this?

Comment (joshaber) [reply app/src/lib/git/update-index.ts]: Are there downsides to the `git rm --cached -r .` approach? I think we should generally steer clear of O(n) operations if at all possible.

Comment (shiftkey) [reply app/src/lib/git/update-index.ts]: Not really, the `--cached` bit is important to ensure you don't touch files in the working directory, but aside from that I'm not aware of any downsides.

Comment (shiftkey) [reply app/src/lib/git/update-index.ts]: It's literally what `git status` suggests, so I'm :thumbsup: to keeping this to one invocation of Git.

**Comment by shiftkey** [comment]
🥉

**Pull request merged**